src/HDInsight/HDInsight/ClusterConfigurationUtils.cs

```diff
@@ -122,5 +127,77 @@ where key.StartsWith(Constants.ClusterConfiguration.StorageAccountKeyPrefix, Str
+        {
+            List<EntraUserInfo> restAuthEntraUsers = new List<EntraUserInfo>();
+            if (String.IsNullOrEmpty(EntraUserIdentity) && (EntraUserFullInfo == null || EntraUserFullInfo.Length == 0))
+            {
+                return restAuthEntraUsers;
+            }
+            if(!string.IsNullOrWhiteSpace(EntraUserIdentity) && (EntraUserFullInfo != null && EntraUserFullInfo.Length > 0))
+            {
+                throw new ArgumentException("Cannot provide both EntraUserIdentity and EntraUserFullInfo parameters.");
+            }
+            if (!string.IsNullOrWhiteSpace(EntraUserIdentity))
+            {
+                try
+                {
+                    var userdata = EntraUserIdentity
+                     .Split(new[] { ',' }, StringSplitOptions.RemoveEmptyEntries)
+                     .Select(s => s.Trim())
+                     .Where(s => !string.IsNullOrEmpty(s))
+                     .ToList();
+                    var graphClient = new GraphServiceClient(new DefaultAzureCredential());
+                    foreach (var data in userdata)
+                    {
+                        try
+                        {
+                            var user = graphClient.Users[data].GetAsync().GetAwaiter().GetResult();
+                            restAuthEntraUsers.Add(new EntraUserInfo
+                            {
+                                ObjectId = user.Id,
+                                DisplayName = user.DisplayName,
+                                Upn = user.UserPrincipalName
+                            });
+                        }
+                        catch (Exception ex)
+                        {
+                            throw new Exception($"Failed to resolve Entra user from input: {data}", ex);
+                        }
+                    }
+                }
+                catch (Exception ex)
+                {
+                    throw new FormatException($"Invalid format in EntraUserIdentity. Expected comma-separated ObjectIds or UPNs, but got: \"{EntraUserIdentity}\"", ex);
+                }
+            }
+            else if (EntraUserFullInfo != null && EntraUserFullInfo.Length > 0)
+            {
+                var userDicts = EntraUserFullInfo.Select(user =>
+                {
+                    var dict = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+                    foreach (DictionaryEntry entry in user)
+                    {
+                        dict[entry.Key.ToString() ?? ""] = entry.Value.ToString() ?? "";
+                    }
+                    return (IDictionary<string, string>)dict;
+                });
+                foreach (var userDict in userDicts)
+                {
+                    string objectId = userDict.TryGetValue("ObjectId", out var oid) ? oid : null;
+                    string upn = userDict.TryGetValue("Upn", out var u) ? u : null;
```

Comment:
so there you required the key is fixed value, you shall write doc for this requirement when define the entrauserfullinfo